src/coreComponents/mesh/MeshManager.cpp

```diff
@@ -61,21 +62,32 @@ void MeshManager::generateMeshes( DomainPartition & domain )
   forSubGroups< MeshGeneratorBase >( [&]( MeshGeneratorBase & meshGen )
   {
     MeshBody & meshBody = domain.getMeshBodies().registerGroup< MeshBody >( meshGen.getName() );
-    meshGen.generateMesh(meshBody); meshBody.createMeshLevel( 0 );
+    meshBody.createMeshLevel( 0 );
+    MeshLevel & meshLevel = meshBody.getBaseDiscretization();
     CellBlockManager & cellBlockManager = meshBody.registerGroup< CellBlockManager >( keys::cellManager );
 
+    meshGen.generateMesh( cellBlockManager );
+    meshBody.setGlobalLengthScale( cellBlockManager.getGlobalLength() );
+    ElementRegionManager & elemManager = meshLevel.getElemManager();
+    for ( WellBlock wellBlock : cellBlockManager.getWellBlocks() ) {
+      WellElementRegion &
+        wellRegion = elemManager.getGroup( ElementRegionManager::groupKeyStruct::elementRegionsGroup() ).
+        getGroup< WellElementRegion >( wellBlock.getWellRegionName() );
+      wellRegion.setWellGeneratorName( wellBlock.getName() );
+      wellRegion.setWellControlsName( wellBlock.getWellControlsName() );
+    }
 
-    MeshGeneratorHelper meshGeneratorHelper = meshGen.generateMesh( meshBody );
-    if ( meshGeneratorHelper.hasMetisNeighborList() )
+    PartitionDescriptor & partitionDescriptor = cellBlockManager.getPartitionDescriptor();
```

Comment:
Do you need a non-`const` version of `PartitionDescriptor`?

Reply:
It was because I was lazy ;).